pkg/controller/wip/controller_test.go

```diff
@@ -31,34 +32,57 @@ import (
 	"k8s.io/kubernetes/pkg/client/testing/core"
 )
 
-func TestReconcileBroker(t *testing.T) {
-	fakeKubeClient, fakeCatalogClient, fakeBrokerCatalog, _, _, testController, _, stopCh := newTestController(t)
+// broker used in most of the tests that need a broker
+var broker = &v1alpha1.Broker{
+	ObjectMeta: v1.ObjectMeta{Name: "test-broker"},
+	Spec: v1alpha1.BrokerSpec{
+		URL:     "https://example.com",
+		OSBGUID: "OSBGUID field",
+	},
+}
 
-	fakeBrokerCatalog.RetCatalog = &brokerapi.Catalog{
-		Services: []*brokerapi.Service{
-			{
-				Name:        "test-service",
-				ID:          "12345",
-				Description: "a test service",
-				Plans: []brokerapi.ServicePlan{
-					{
-						Name:        "test-plan",
-						Free:        true,
-						ID:          "34567",
-						Description: "a test plan",
-					},
+var serviceClass = &v1alpha1.ServiceClass{
+	ObjectMeta: v1.ObjectMeta{Name: "test-serviceclass"},
+	BrokerName: "test-broker",
+	Plans: []v1alpha1.ServicePlan{{
+		Name:    "default",
+		OSBFree: true,
+		OSBGUID: "9876543",
+	}},
+}
+
+var testCatalog = &brokerapi.Catalog{
+	Services: []*brokerapi.Service{
+		{
+			Name:        "test-service",
+			ID:          "12345",
+			Description: "a test service",
+			Plans: []brokerapi.ServicePlan{
+				{
+					Name:        "test-plan",
+					Free:        true,
+					ID:          "34567",
+					Description: "a test plan",
 				},
 			},
 		},
-	}
+	},
+}
 
-	broker := &v1alpha1.Broker{
-		ObjectMeta: v1.ObjectMeta{Name: "test-name"},
-		Spec: v1alpha1.BrokerSpec{
-			URL:     "https://example.com",
-			OSBGUID: "OSBGUID field",
-		},
-	}
+type instanceParameters struct {
+	Name string            `json:"name"`
+	Args map[string]string `json:"args"`
+}
+
+type bindingParameters struct {
+	Name string   `json:"name"`
+	Args []string `json:"args"`
+}
+
+func TestReconcileBroker(t *testing.T) {
+	fakeKubeClient, fakeCatalogClient, fakeBrokerCatalog, _, _, testController, _, stopCh := newTestController(t)
```

Comment:
can you call `defer close(stopCh)`, instead of at the end of the test. same with the other tests

not blocking for this PR, however, since it's in many other tests. If you decide not to do it, can you create an issue for it?